ompi/runtime/ompi_mpi_init.c

```diff
@@ -268,6 +269,50 @@ MPI_Fint *MPI_F08_STATUSES_IGNORE = NULL;
 
 #include "mpif-c-constants.h"
 
+static const char *ompi_thread_level_keywords[] = {"single", "serialized", "funneled", "multiple"};
+static const char *ompi_thread_level_prepositions[] = {"mpi_thread_", "thread_", NULL};
+/* In the future integer MPI_ABI values for MPI_THREAD_SINGLE-MULTIPLE may be
+ * non-sequential (but ordered) integer values. If you are implementing MPI
+ * ABI changes please refer to
+ * https://github.com/open-mpi/ompi/pull/13211#discussion_r2085086844
+ */
+static const int ompi_thread_level_values[] = {MPI_THREAD_SINGLE, MPI_THREAD_SERIALIZED,
+                                               MPI_THREAD_FUNNELED, MPI_THREAD_MULTIPLE};
+
+int ompi_getenv_mpi_thread_level(int *requested)
+{
+    char* env;
+    if (NULL != (env = getenv("OMPI_MPI_THREAD_LEVEL"))) {
+        char *prep = NULL, *token = (char *) env /* full match */;
+        int pidx = 0, found = strtol(env, &prep, 10);
```

Comment:
Using strtol instead of strcmp will match `1abcd` which is technically an invalid input.